src/lightning_app/core/flow.py

```diff
@@ -149,6 +150,8 @@ def __setattr__(self, name, value):
             elif isinstance(value, LightningWork):
                 self._works.add(name)
                 _set_child_name(self, value, name)
+                # Question: I do not understand, why are we removing it from the state
```

Comment:
We aren't removing it from the state. We are simply naming the component as {parent_name}.{attribute_name}.

Reply:
yes, that is happening inside `_set_child_name`, but I guess the name is removed from the state when an existing attribute is set to Flow/Work. But I don't think this is happening when it's set to any of the structures.